### PR TITLE
Refactoring old types

### PR DESCRIPTION
**Pull request by lmoe**
# Description of change

Please write a summary of your changes and why you made them. 

## Links to any relevant issues

Be sure to reference any related issues by adding `fixes issue #`.

## Type of change

Choose a type of change, and delete any options that are not relevant.

- Bug fix (a non-breaking change which fixes an issue)
- Enhancement (a non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- Documentation Fix

## How the change has been tested

Describe the tests that you ran to verify your changes.

Make sure to provide instructions for the maintainer as well as any relevant configurations.

## Change checklist

Tick the boxes that are relevant to your changes, and delete any items that are not.

- [ ] I have followed the [contribution guidelines](/CONTRIBUTING.md) for this project
- [ ] I have performed a self-review of my own code
- [ ] I have selected the `develop` branch as the target branch
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that new and existing unit tests pass locally with my changes

### CODE REVIEW

File: clients/iscmove/isc.go [hunk 1]
```diff
@@ -90,7 +90,7 @@ type AssetsBagWithBalances struct {
 
 type Anchor struct {
 	ID         sui.ObjectID
-	Assets     Referent[AssetsBag]
+	Assets     Referent[AssetsBagWithBalances]
```
Comment (dessaya) [inline clients/iscmove/isc.go:93]: Why this change? This doesn't break the BCS decoding?

File: clients/iscmove/iscmoveclient/client_request_test.go [hunk 2]
```diff
@@ -63,6 +63,7 @@ func TestCreateAndSendRequest(t *testing.T) {
 	)
 	require.NoError(t, err)
 
-	_, err = createAndSendRequestRes.GetCreatedObjectInfo(iscmove.RequestModuleName, iscmove.RequestObjectName)
-	require.NoError(t, err)
+	mod, err := createAndSendRequestRes.GetCreatedObjectInfo(iscmove.RequestModuleName, iscmove.RequestObjectName)
+	mod.
+		require.NoError(t, err)
```
Comment (dessaya) [inline clients/iscmove/iscmoveclient/client_request_test.go:68]: This doesn't look right :thinking:

**Review by dessaya — APPROVED** [review]

File: clients/iscmove/isc.go (hunk 1 above)
Comment (lmoe) [reply clients/iscmove/isc.go:93]: I believe it doesn't but this needs some testing later. I changed this, because we use this specific Anchor type now almost everywhere and in certain parts we require the balance of the Anchor that we don't comfortably have without the WithBalances type. The State update for example uses those. We can't convert an AssetBag to isc.Assets, only an AssetBagWithBalance. 

Otherwise we have to pull in the L1 client into the VM RequestContxext parts and pull it every time. 

But open for discussion to make it better. Imo it should not interfere with BCS because AssetsBagWithBalance is including AssetsBag+parsed coins. So the initial AssetBag would be filled regardless. But - yeah, needs verification. For now I just wanted to make some progress to make this project compilable again.

File: clients/iscmove/iscmoveclient/client_request_test.go (hunk 2 above)
Comment (lmoe) [reply clients/iscmove/iscmoveclient/client_request_test.go:68]: yup :D
